client/my-sites/customer-home/cards/tasks/site-setup-list/mobile-app-download.jsx

```diff
@@ -0,0 +1,48 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { translate } from 'i18n-calypso';
+import AppsBadge from 'calypso/blocks/get-apps/apps-badge';
+import userAgent from 'calypso/lib/user-agent';
+
+const MobileAppDownload = () => {
+	const { isiPad, isiPod, isiPhone, isAndroid } = userAgent;
+	const isIos = isiPad || isiPod || isiPhone;
+
+	const showBadge = ( isIosDevice, isAndroidDevice ) => {
```

Comment:
This arrow function doesn't depend on anything within the `MobileAppDownload` function, so you should be able to hoist it up into it's own standalone function. This will prevent the function being re-created over and over again each time `MobileAppDownload` is rendered.

You don't need to hoist it out to it's own component with a capital letter and everything, it's enough to just pull it out as a helper function.

Reply:
This should be fixes.